### PR TITLE
Update French pattern

### PR DESCRIPTION
**Pull request by cramoisi**
for the word désir
revert 380 and edit 1886 and 1887

`<pattern> dé3s2i3r</pattern>`
`<pattern> dé3s2i4r </pattern>`
`<pattern> dé3s2i4rs </pattern>`

### CODE REVIEW

**Comment by cramoisi** [comment]
@poire-z this shoud do the trick. I did a mistake in my two last rules months ago and I'm amazed it only emerge now.

**Comment by cramoisi** [comment]
(yeah, I switched 2 and 4 in my comment)

**Comment by poire-z** [comment]
Looks like with this, I get again:
```
dire du dési-
r de certains
```

**Comment by cramoisi** [comment]
@poire-z  : could you switch  2 and 4 to get them as in my first comment and test it again ?

**Comment by poire-z** [comment]
Yep, that works better :)

Not super clear from the code, but so, I guess it works like this?:
`<pattern> dé3s2i3r</pattern>` is matched, so `de-si-r` could work
`<pattern> dé3s2i4r </pattern>` is later matched, so `de-siXr` and the 2nd hyphen is cancelled ? Because of the 4 which is greater than 3? it would not cancel it if you had used `dé3s2i2r` ?

**Comment by cramoisi** [comment]
@poire-z : yep, it works like that :)  Could you make the change in the PR, I did it yest inside the github website and I can't add more commit when I do dirty work like that :/

File: cr3gui/data/hyph/French.pattern [hunk 1]
```diff
@@ -1883,7 +1883,7 @@ but needed by crengine no quote pattern -->
 <pattern>n2x </pattern>
 <pattern>o2s </pattern>
 <!-- other rules -->
-<pattern> dé3s5i2r </pattern>
-<pattern> dé3s5i2rs </pattern>
+<pattern> dé3s4i2r </pattern>
```
Comment (poire-z) [inline cr3gui/data/hyph/French.pattern]:
```suggestion
<pattern> dé3s2i4r </pattern>
```

File: cr3gui/data/hyph/French.pattern [hunk 2]
```diff
@@ -1883,7 +1883,7 @@ but needed by crengine no quote pattern -->
...
+<pattern> dé3s4i2rs </pattern>
```
Comment (poire-z) [inline cr3gui/data/hyph/French.pattern]:
```suggestion
<pattern> dé3s2i4rs </pattern>
```

**Comment by poire-z** [comment]
>  Could you make the change in the PR

Trying that with "suggestions": you need to go to Files changes tab and accept my suggestions (if I didn't mess them:) : that will make 1 or 2 more commits that I will squash into a single one.

**Comment by cramoisi** [comment]
@poire-z : it looks like it's finally right :) thanks !

**Comment by cramoisi** [comment]
remindeder : these rules only exists because the min-right-hyph need to be 1 within crengine to work with some French words. The rules in the tex-hyph files are made to work with min-right-hyph at 2.

**Pull request merged**